### PR TITLE
Update to 3.0

### PR DESCRIPTION
**Pull request by dopplershift**
Rather than merge, which had a bunch of conflicts, I've just grabbed the recipe files, patches, and config. I'll use the bot to re-render.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dopplershift** [comment]
Ok, so Circle worked, but looking through the logs it only did Python 3.6 and Python 3.7. It *seems* like it's no longer using separate CI jobs for the different versions (ok, fine), but WTH is the Python 2.7 build?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,88 +1,79 @@
...
     # Find libpng on Windows.
     - setupext.patch  # [win]
-    # A patch to make Qt the default backend.
-    - cfg_qt5agg.patch  # [not osx]
-    # Removes framework check from osx backend (seems unneeded)
-    - osx-frame.patch  # [osx]
 
 build:
   number: 200
+  skip: true  # [py<35]
```
Comment (dougalsutherland) [inline recipe/meta.yaml]: This might have something to do with why there's no Python 2.7 build.... :)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,88 +1,79 @@
...
 
 requirements:
```
Comment (dougalsutherland) [inline recipe/meta.yaml]: Maybe you need a python dep here (and not just under the output) for the CI jobs to run separately? Not sure.

**Comment by dopplershift** [comment]
:sheep: Well that’s embarrassing—forgot that matplotlib 3 is Python 3 only.

**Comment by QuLogic** [comment]
Bump for 3.0.1?

**Comment by dopplershift** [comment]
Already have, but Circle is broken. Thinking it’s due to the weird build matrix we have here after re-rendering, which happened after I added `conda_build_config.yml`.

**Comment by QuLogic** [comment]
It seems to be testing 3.6 packages against 3.7?
```
TEST START: /home/conda/feedstock_root/build_artifacts/linux-64/matplotlib-base-3.0.1-py36hc039c98_200.tar.bz2

## Package Plan ##

  environment location: /home/conda/feedstock_root/build_artifacts/matplotlib_1540526538341/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho


The following NEW packages will be INSTALLED:

    blas:            1.1-openblas                        conda-forge
    bzip2:           1.0.6-h470a237_2                    conda-forge
    ca-certificates: 2018.10.15-ha4d7672_0               conda-forge
    certifi:         2018.10.15-py37_1000                conda-forge
```
test start is `py36`, certifi is `py37`.

File: .ci_support/linux_.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- condaforge/linux-anvil
+freetype:
+- 2.9.1
+icu:
+- '58'
+libpng:
+- 1.6.34
+pin_run_as_build:
+  freetype:
+    max_pin: x
```
Comment (QuLogic) [inline .ci_support/linux_.yaml]: Do we need freetype pinning on this build?

**Comment by ocefpaf** [comment]
@dopplershift you may need the changes in #186 to get this one working.

**Comment by dopplershift** [comment]
Thanks for the pointer @ocefpaf .

**Pull request merged**